src/service/viewer-impl.js

```diff
@@ -499,6 +511,34 @@ export class Viewer {
     return this.isWebviewEmbedded_;
   }
 
+  /**
+   * Whether the document is using Chrome Custom Tabs.
+   * @return {boolean}
+   */
+  isCustomTab() {
+    return this.isCustomTab_;
+  }
+
+  /**
+   * Update the URL fragment with data needed to support custom tabs.
+   */
+  maybeUpdateFragmentForCustomTab() {
+    if (!this.isCustomTab()) {
+      return;
+    }
+    if (!this.win.history.replaceState) {
```

Comment:
Do all CCT browsers support history? If not then it should fall back to setting `window.location.hash`. 

https://caniuse.com/#feat=history

Reply:
Looks like any version of Chrome can be embedded in a tab. So, it is possible that the history API won't be present (just very unlikely). I'll fallback to window.location.hash.

Reply:
Actually, it looks like CCT only works with Chrome 45 and newer. So I'll keep the check, but if it's not supported, simply do nothing (this shouldn't happen but is safe and will help us avoid unnecessarily adding the ampshare param).